congen CLI wedged at startup: ExceptionInInitializerError from json-schema-validator v3 dialect mismatch

## Describe the Bug

The `congen` CLI is wedged at startup. Every entry point — including `congen -h`, `congen list`, and `congen scan` — crashes before producing output. The only thing keeping this from being visible is that nobody on CI exercises the appassembler binary; unit tests bypass the CLI and call generators directly.

```
$ ./element-template-generator/congen-cli/target/appassembler/bin/congen -h
Exception in thread "main" java.lang.ExceptionInInitializerError
    at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized0(Native Method)
    ...
    at picocli.CommandLine$Model$CommandReflection.initSubcommands(CommandLine.java:11908)
    at picocli.CommandLine$Model$CommandSpec.forAnnotatedObject(CommandLine.java:6396)
    at picocli.CommandLine.<init>(CommandLine.java:230)
    at io.camunda.connector.generator.cli.Main.main(Main.java:25)
Caused by: com.networknt.schema.InvalidSchemaException:
    Unknown dialect 'http://json-schema.org/draft-07/schema#'.
    Only dialects that are explicitly configured can be used.
    at com.networknt.schema.dialect.BasicDialectRegistry.getDialect(BasicDialectRegistry.java:61)
    at com.networknt.schema.SchemaRegistry.getDialectOrDefault(SchemaRegistry.java:573)
    at com.networknt.schema.SchemaRegistry.getMappedSchema(SchemaRegistry.java:780)
    at com.networknt.schema.SchemaRegistry.loadSchema(SchemaRegistry.java:754)
    at com.networknt.schema.SchemaRegistry.getSchema(SchemaRegistry.java:669)
    at io.camunda.connector.generator.cli.command.Generate.<clinit>(Generate.java:44)
```

## Steps to Reproduce

1. `./mvnw clean package -pl element-template-generator/congen-cli -am -DskipTests`
2. `./element-template-generator/congen-cli/target/appassembler/bin/congen -h`

Or in CI, observe the new smoke test in #7354 fail:
- **Failing CI run**: https://github.com/camunda/connectors/actions/runs/26450331753/job/77868959986
- **PR with regression test**: #7354

## Expected Behavior

The CLI should print help, list generators, scan a spec, or generate a template depending on the subcommand — anything except crash at startup.

## Root Cause

PR #5923 (`fix(deps): update dependency com.networknt:json-schema-validator to v3`, merged 2025-12-15) bumped the schema validator 2.x → 3.x. networknt 3.x removed the previously-implicit registration of older JSON Schema draft dialects: dialects must now be explicitly registered before they can be referenced.

`element-template-generator/congen-cli/src/main/java/io/camunda/connector/generator/cli/command/Generate.java:40-46` constructs a `SchemaRegistry` with only draft-2020-12:

```java
static final Schema jsonSchema =
    SchemaRegistry.withDialect(
            Dialects.getDraft202012(),
            builder -> builder.schemaLoader(Builder::fetchRemoteResources))
        .getSchema(
            SchemaLocation.of(
                "https://unpkg.com/@camunda/zeebe-element-templates-json-schema/resources/schema.json"));
```

…then fetches the Camunda element-template schema, which declares `"$schema": "http://json-schema.org/draft-07/schema#"`. The registry can't resolve draft-07, so the load throws an `InvalidSchemaException`.

This is a `static final` initializer in `Generate`. picocli reflectively instantiates every subcommand class when the root `CommandLine` is constructed (`CommandReflection.initSubcommands` in the stack trace), triggering the static initializer regardless of which subcommand the user actually invoked. Result: every entry point crashes, including `-h`.

## Why nobody noticed for ~5 months

- Unit tests call `OpenApiOutboundTemplateGenerator` and other generator classes directly. They never construct the picocli command tree, so the static initializer never runs.
- The appassembler binary is built in CI as a side effect of `mvn package`, but nothing in CI actually executes it.
- No tests existed under `element-template-generator/congen-cli/` at all — the smoke test in #7354 is the first.

## Proposed Fix

Minimal: swap the dialect to draft-7 to match the schema being loaded.

```diff
- Dialects.getDraft202012(),
+ Dialects.getDraft7(),
```

`Dialects.getDraft7()` is part of networknt 3.x. The validator module already uses the equivalent draft-07 selector (`JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V7)`) without trouble.

A slightly broader fix would register both dialects via `SchemaRegistry.withDialects(List.of(Dialects.getDraft7(), Dialects.getDraft202012()), ...)` so future schema-version upgrades don't introduce the same bug. Worth considering as part of this fix or a follow-up.

A more robust separable fix is to **lazy-initialize the schema** in `Generate.call()` rather than in a `static final` initializer. Then a broken or unreachable schema only fails the `generate` command, not every other subcommand. Recommended as a follow-up but not required for this bug.

## Environment

- Library version: 8.10.0-SNAPSHOT (main as of `0ee23b1c`)
- networknt:json-schema-validator: 3.0.2
- Java: 21

--- a/element-template-generator/congen-cli/pom.xml
+++ b/element-template-generator/congen-cli/pom.xml
@@ -61,6 +61,17 @@
       <groupId>org.slf4j</groupId>
       <artifactId>slf4j-nop</artifactId>
     </dependency>
+
+    <dependency>
+      <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.assertj</groupId>
+      <artifactId>assertj-core</artifactId>
+      <scope>test</scope>
+    </dependency>
   </dependencies>
 
   <build>

--- a/element-template-generator/congen-cli/src/main/java/io/camunda/connector/generator/cli/command/Generate.java
+++ b/element-template-generator/congen-cli/src/main/java/io/camunda/connector/generator/cli/command/Generate.java
@@ -39,8 +39,7 @@ public class Generate implements Callable<Integer> {
 
   static final Schema jsonSchema =
       SchemaRegistry.withDialect(
-              Dialects.getDraft202012(),
-              builder -> builder.schemaLoader(Builder::fetchRemoteResources))
+              Dialects.getDraft7(), builder -> builder.schemaLoader(Builder::fetchRemoteResources))
           .getSchema(
               SchemaLocation.of(
                   "https://unpkg.com/@camunda/zeebe-element-templates-json-schema/resources/schema.json"));

--- a/element-template-generator/congen-cli/src/test/java/io/camunda/connector/generator/cli/CliBootstrapTest.java
+++ b/element-template-generator/congen-cli/src/test/java/io/camunda/connector/generator/cli/CliBootstrapTest.java
@@ -1,0 +1,47 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.generator.cli;
+
+import static org.assertj.core.api.Assertions.assertThatCode;
+
+import io.camunda.connector.generator.cli.command.ConGen;
+import org.junit.jupiter.api.Test;
+import picocli.CommandLine;
+
+/**
+ * Smoke tests guarding against startup-time crashes in the CLI. picocli reflects over every
+ * subcommand class when the root {@link CommandLine} is constructed, so any failing {@code static}
+ * initializer in a subcommand wedges every entry point — including {@code -h}, {@code list}, and
+ * {@code scan} that don't touch the failing class otherwise.
+ */
+class CliBootstrapTest {
+
+  @Test
+  void commandLineCanBeConstructed() {
+    assertThatCode(() -> new CommandLine(new ConGen())).doesNotThrowAnyException();
+  }
+
+  @Test
+  void helpFlagReturnsZeroExitCode() {
+    int exitCode =
+        new CommandLine(new ConGen()).setUnmatchedOptionsArePositionalParams(true).execute("-h");
+    assertThatCode(() -> {}).doesNotThrowAnyException();
+    if (exitCode != 0) {
+      throw new AssertionError("expected -h to exit 0, got " + exitCode);
+    }
+  }
+}

